src/coord/coord.rs

```diff
@@ -2037,12 +2086,11 @@ where
                     // reduce the volume of the collection, but we don't have that
                     // information here.
                     if !index_state.upper.frontier().is_empty() {
```

Comment:
is the empty `upper` frontier an indicator that the collection is complete / has seen all of its inputs and dropped its capability?

Reply:
That is correct.